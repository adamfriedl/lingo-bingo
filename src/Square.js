// @flow

import React from 'react';
import { Flex, Text } from 'rebass';
import styled from 'styled-components';

const CustomFlex = styled(Flex)`
  float: left;
  width: 17%;
  border: 5px solid #07c;

  :after {
    content: '';
    display: inline;
    padding-bottom: 100%;
  }
`;

const Square = (props: {
  text: string,
  clicked: boolean,
  onClick: () => void
}) => (
  <div>
    <CustomFlex
      align="center"
      justify="center"
      bg={props.clicked ? 'yellow' : 'white'}
      mr={1}
      mb={1}
      onClick={props.onClick}
    >
      <Flex wrap>
        <Text f={[1, 2, 3]} p={1} center color="blue">
          {props.text}
        </Text>
      </Flex>
    </CustomFlex>
  </div>
);

export default Square;
