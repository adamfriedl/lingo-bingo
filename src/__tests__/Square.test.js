import React from 'react';
import { shallow } from 'enzyme';
import Square from '../Square';

it('renders', () => {
  const component = shallow(<Square />);
  expect(component).toMatchSnapshot();
});
