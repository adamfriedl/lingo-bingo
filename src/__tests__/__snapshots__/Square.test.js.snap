// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Square />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Styled(Flex)
        align="center"
        bg="white"
        justify="center"
        mb={1}
        mr={1}
        onClick={undefined}
>
        <Flex
                wrap={true}
        >
                <Text
                        center={true}
                        color="blue9"
                        f={
                                Array [
                                        1,
                                        2,
                                        3,
                                      ]
                        }
                        m={0}
                        p={1}
                />
        </Flex>
</Styled(Flex)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "align": "center",
        "bg": "white",
        "children": <Flex
          wrap={true}
>
          <Text
                    center={true}
                    color="blue9"
                    f={
                              Array [
                                        1,
                                        2,
                                        3,
                                      ]
                    }
                    m={0}
                    p={1}
          />
</Flex>,
        "justify": "center",
        "mb": 1,
        "mr": 1,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            center={true}
            color="blue9"
            f={
                        Array [
                                    1,
                                    2,
                                    3,
                                  ]
            }
            m={0}
            p={1}
/>,
          "wrap": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "center": true,
            "children": undefined,
            "color": "blue9",
            "f": Array [
              1,
              2,
              3,
            ],
            "m": 0,
            "p": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Styled(Flex)
          align="center"
          bg="white"
          justify="center"
          mb={1}
          mr={1}
          onClick={undefined}
>
          <Flex
                    wrap={true}
          >
                    <Text
                              center={true}
                              color="blue9"
                              f={
                                        Array [
                                                  1,
                                                  2,
                                                  3,
                                                ]
                              }
                              m={0}
                              p={1}
                    />
          </Flex>
</Styled(Flex)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "align": "center",
          "bg": "white",
          "children": <Flex
            wrap={true}
>
            <Text
                        center={true}
                        color="blue9"
                        f={
                                    Array [
                                                1,
                                                2,
                                                3,
                                              ]
                        }
                        m={0}
                        p={1}
            />
</Flex>,
          "justify": "center",
          "mb": 1,
          "mr": 1,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              center={true}
              color="blue9"
              f={
                            Array [
                                          1,
                                          2,
                                          3,
                                        ]
              }
              m={0}
              p={1}
/>,
            "wrap": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "center": true,
              "children": undefined,
              "color": "blue9",
              "f": Array [
                1,
                2,
                3,
              ],
              "m": 0,
              "p": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
